chain/txValidator.go

```diff
@@ -264,107 +285,199 @@ func VerifyTransactionWithLedger(txn *transaction.Transaction) error {
 	return nil
 }
 
-type Iterator interface {
-	Iterate(handler func(item *transaction.Transaction) error) error
+type subscription struct {
+	topic      string
+	bucket     uint32
+	subscriber Uint160
+	identifier string
+}
+
+type BlockValidationState struct {
+	txnlist           map[Uint256]struct{}
```

Comment:
I think all theses maps needs to be sync.Map because append to txn pool is called concurrently (which calls process txn which calls verify txn with block)

Reply:
Also sync.Map LoadOrStore would be quite good for our atomic verification purpose

Reply:
done